examples/mongocxx/mongodb.com/documentation_examples.cpp

```diff
@@ -1368,6 +1485,113 @@ static bool version_at_least(mongocxx::v_noabi::database& db, int minimum_major)
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_array;
+    using bsoncxx::builder::basic::make_document;
+    using bsoncxx::types::bson_value::value;
+
+    // Drop data from prior test runs.
+    client["keyvault"]["datakeys"].drop();
+    client["docsExamples"].drop();
+
+    // Create two data keys.
+    class client key_vault_client {
+        uri{}, add_test_server_api(),
+    };
+
+    options::client_encryption ce_opts;
+    ce_opts.key_vault_client(&key_vault_client);
+    ce_opts.key_vault_namespace({"keyvault", "datakeys"});
+    ce_opts.kms_providers(_make_kms_doc(false));
+    client_encryption client_encryption(std::move(ce_opts));
+
+    auto key1_id = client_encryption.create_data_key("local");
+    auto key2_id = client_encryption.create_data_key("local");
+
+    // Create an encryptedFieldsMap.
+    auto encrypted_fields_map = make_document(kvp(
+        "docsExamples.encrypted",
+        make_document(kvp(
+            "fields",
+            make_array(make_document(kvp("path", "encryptedIndexed"),
+                                     kvp("bsonType", "string"),
+                                     kvp("keyId", key1_id),
+                                     kvp("queries", make_document(kvp("queryType", "equality")))),
+                       make_document(kvp("path", "encryptedUnindexed"),
+                                     kvp("bsonType", "string"),
+                                     kvp("keyId", key2_id)))))));
+
+    // Create an FLE 2 collection.
+    options::auto_encryption auto_encrypt_opts{};
+    auto_encrypt_opts.key_vault_namespace({"keyvault", "datakeys"});
+    auto_encrypt_opts.kms_providers(_make_kms_doc(false));
+    auto_encrypt_opts.encrypted_fields_map(encrypted_fields_map.view());
+
+    // Optional, If mongocryptd is not in PATH, then find the binary at MONGOCRYPTD_PATH.
+    char* mongocryptd_path = std::getenv("MONGOCRYPTD_PATH");
+    if (mongocryptd_path) {
+        auto_encrypt_opts.extra_options(
+            make_document(kvp("mongocryptdSpawnPath", mongocryptd_path)));
+    }
+
+    mongocxx::options::client encrypted_client_opts;
+    encrypted_client_opts.auto_encryption_opts(std::move(auto_encrypt_opts));
+    class client encrypted_client {
+        uri{}, add_test_server_api(encrypted_client_opts)
+    };
+
+    // Create the FLE 2 collection docsExample.encrypted.
+    // Because docsExample.encrypted is in encryptedFieldsMap, it is created with FLE 2 support.
+    auto db = encrypted_client["docsExamples"];
+    db.create_collection("encrypted");
+    auto encrypted_collection = db["encrypted"];
+
+    // Auto encrypt an insert and find.
+    {
+        // Encrypt an insert.
+        encrypted_collection.insert_one(make_document(kvp("_id", 1),
+                                                      kvp("encryptedIndexed", "indexedValue"),
+                                                      kvp("encryptedUnindexed", "unindexedValue")));
```

Comment:
Attempting to run this test yields the following error on this line:
```
terminate called after throwing an instance of 'mongocxx::v_noabi::bulk_write_exception'
  what():  mongocryptd error: jsonSchema is a required command field:: generic server error
Aborted
```
@kevinAlbs Is this field also missing in the C Driver and Go Driver reference examples?

Reply:
Here's the link for the corresponding lines from the Go driver ticket: https://github.com/mongodb/mongo-go-driver/pull/969/files#diff-35d11cdcb894289a5379a0fb50235f1933e872f4f591cfd2583245f43f4ee8fcR551-R558

Reply:
I haven't seen this error on my machine or in CI, on which platform and server version are you seeing this error?

Reply:
Ubuntu 20.04 (in WSL) with MongoDB Server 6.1.0 using a 3-server replica set topology.

Reply:
That's interesting. I don't see the field 'jsonSchema' in either the Go or C drivers (link to the corresponding lines in the C driver for reference: https://github.com/mongodb/mongo-c-driver/commit/a74437ea56361b596c0e90cbe446c703611b002b#diff-d363a5539f2fbabfadf2d7cc6cdb96a203fb63f2caa31bae731c3e12d97b204cR4675-R4697)

Reply:
That's a relief 😄